e2e/epinio.e2e.spec.ts

```diff
@@ -0,0 +1,239 @@
+ * Helper to identify the Load Balancer IP Address.
+ * It will return the traefik IP address, required by epinio install.
+ */
+export async function loadBalancerIp() {
+  const serviceInfo = await kubectl('describe', 'service', 'traefik', '--namespace', 'kube-system');
+  const serviceFiltered = serviceInfo.split('\n').toString();
+  const m = /LoadBalancer Ingress:\s+(((?:[0-9]{1,3}\.){3}[0-9]{1,3}))/.exec(serviceFiltered);
+
+  // checking if it will be undefined, null, 0 or empty
+  if (m) {
+    return m[1];
+  } else {
+    console.log('Cannot find load balancer IP address.');
+  }
+}
+
+const platforms: Record<string, string> = { darwin: 'darwin', win32: 'win32', linux: 'linux' };
+
+export async function installEpinioCli() {
+  const platform = os.platform() as string;
+
+  if (!platforms[platform]) {
+    console.error(`Platform type not detect. Found: ${ platform }`);
+  }
+
+  // Download epinio binary based on platform type
+  await downloadEpinioBinary(platform);
+}
+
+/**
+ * Download epinio binary based on the platform type and save the binary
+ * into a temporary folder.
+ */
+export async function downloadEpinioBinary( platformType: string) {
+  // Setting up epinio binaries names per platform
+  const epinioWin = 'epinio-windows-amd64.exe';
+  const epinioLinux = 'epinio-linux-x86_64';
+  const epinioDarwin = 'epinio-darwin-x86_64';
+  const epinioDarwinArm = 'epinio-darwin-arm64';
+  const epinioWorkingVersion = 'v0.5.0';
+
+  // Create a temp folder for epinio binary
+  const epinioTempFolder = path.join(os.homedir(), 'epinio-tmp');
+
+  if (!fs.existsSync(epinioTempFolder)) {
+    fs.mkdirSync(epinioTempFolder, { recursive: true });
+  }
+
+  // Detect CPU arch
+  const cpuArch = os.arch();
+
+  switch (platformType) {
+  case 'darwin':
+    if (cpuArch === 'x64') {
+      await downloadEpinioCommand(epinioWorkingVersion, epinioDarwin, epinioTempFolder);
+      break;
+    } else {
+      await downloadEpinioCommand(epinioWorkingVersion, epinioDarwinArm, epinioTempFolder);
+      break;
+    }
+  case 'linux':
+    await downloadEpinioCommand(epinioWorkingVersion, epinioLinux, epinioTempFolder);
+    break;
+  case 'win32':
+    await downloadEpinioCommand(epinioWorkingVersion, epinioWin, epinioTempFolder);
+    break;
+  }
+}
+
+/**
+ * Download latest epinio cli binary and makes it executable
+ */
+export async function downloadEpinioCommand(version: string, platform: string, folder: string) {
+  const epinioUrl = 'https://github.com/epinio/epinio/releases/download/';
+
+  if (!os.platform().startsWith('win')) {
+    await curl('--fail', '--location', `${ epinioUrl }${ version }/${ platform }`, '--output', `${ folder }\/epinio`);
+    const stat = fs.statSync(`${ folder }\/epinio`).mode;
+
+    fs.chmodSync(`${ folder }\/epinio`, stat | 0o755);
+  } else {
+    const winPath = path.resolve(folder);
+    await curl('--fail', '--location', `${ epinioUrl }${ version }/${ platform }`, '--output', `${ winPath }\\epinio.zip`);
```

Comment:
Lint issues.

Should we use `fetch` instead?

Reply:
It makes the code a little more elegant.